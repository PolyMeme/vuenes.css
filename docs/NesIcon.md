# nes-icon 

## props 

- `name` ***String*** (*required*) 

- `medium` ***Boolean*** (*optional*) 

- `large` ***Boolean*** (*optional*) 

- `empty` ***Boolean*** (*optional*) 

## computed properties 

- `iconClasses` 

   **dependencies:** `name`, `medium`, `large`, `empty` 


