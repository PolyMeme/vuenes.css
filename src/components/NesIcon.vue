<template>
  <li
    :class="iconClasses"
    class="nes-icon"
  />
</template>

<script>
/**
 * Vue Component implementation of the Nes Icons
 * @author Jesus Fernando Alvarez Franco
 * @license MIT
 */
export default {
  props: {
    /**
     * Define the type of icon to be used
     */
    name: {
      type: String,
      required: true
    },
    /**
     * Defines if the icon is small by using the `is-small` css class
     * @default false
     */
    small: Boolean,
    /**
     * Defines if the icon is medium by using the `is-medium` css class
     * @default false
     */
    medium: Boolean,
    /**
     * Defines if the icon is large by using the `is-large` css class
     * @default false
     */
    large: Boolean,
    /**
     * Defines if the icon is empty by using the `is-empty` css class
     * @default false
     */
    empty: Boolean,
    /**
     * Defines if the icon is half by using the `is-half` css class
     * @default false
     */
    half: Boolean
  },
  computed: {
    /**
     * Calculate the css classes to be used for the icon depending of the props
     * @returns {{'is-half': BooleanConstructor, 'is-medium': BooleanConstructor, 'is-large': BooleanConstructor, 'is-small': BooleanConstructor, 'is-empty': BooleanConstructor}}
     */
    iconClasses () {
      return {
        [this.name]: true,
        'is-small': this.small,
        'is-medium': this.medium,
        'is-large': this.large,
        'is-empty': this.empty,
        'is-half': this.half
      }
    }
  }
}
</script>
