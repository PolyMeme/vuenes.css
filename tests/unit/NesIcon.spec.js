import { mount } from '@vue/test-utils'
import NesIcon from '../../src/components/NesIcon.vue'

describe('NesIcon.vue', () => {
  const createIconWrapper = (propsData = {}) => {
    return mount(NesIcon, {
      propsData
    })
  }

  it('should render icon properly', () => {
    const wrapper = createIconWrapper({
      name: 'like'
    })

    expect(wrapper.find('li.nes-icon.like').exists()).toBe(true)
  })

  it('should render icon medium', () => {
    const wrapper = createIconWrapper({
      name: 'like',
      medium: true
    })

    expect(wrapper.find('li.nes-icon.like.is-medium').exists()).toBe(true)
  })

  it('should render icon large', () => {
    const wrapper = createIconWrapper({
      name: 'like',
      large: true
    })

    expect(wrapper.find('li.nes-icon.like.is-large').exists()).toBe(true)
  })

  it('should render icon empty', () => {
    const wrapper = createIconWrapper({
      name: 'like',
      empty: true
    })

    expect(wrapper.find('li.nes-icon.like.is-empty').exists()).toBe(true)
  })

  it('should match snapshot', () => {
    const wrapper = createIconWrapper({
      name: 'like',
      large: true,
      empty: true
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
