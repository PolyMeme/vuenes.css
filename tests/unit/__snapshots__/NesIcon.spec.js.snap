// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesIcon.vue should match snapshot 1`] = `<li class="nes-icon like is-large is-empty"></li>`;
